types/address.go

```diff
@@ -78,11 +78,11 @@ const (
 var (
 	// AccAddress.String() is expensive and if unoptimized dominantly showed up in profiles,
```

Comment:
It's unforunate we have to rely on singletons for the cache and mutexes :-/

Reply:
True. Changing it would involve a big refactoring -> reducing package public functions and making everything managed by objects.